perllib/FixMyStreet/App/Controller/Reports.pm

```diff
@@ -85,6 +77,60 @@ sub index : Path : Args(0) {
     $c->response->header('Cache-Control' => 'max-age=3600');
 }
 
+=head2 get_and_stash_bodies
+
+Gets and stashes the body list, honouring mapit_type and mapit_container
+if provided.
+
+=cut
+
+sub get_and_stash_bodies {
+    my ($self, $c) = @_;
+    my $rs = $c->model('DB::Body')->search({}, {
+        '+select' => [ { count => 'area_id' } ],
+        '+as' => [ 'area_count' ],
+        join => 'body_areas',
+        distinct => 1,
+    });
+
+    my @types = $c->req->param('type');
+
+    $c->stash->{report_filter} = 'all';
+    my $areas = do {
+        if (my $parent = $c->req->param('parent')) {
+            @types = @types ? @types : @{ $c->cobrand->area_types };
+
+            mySociety::MaPit::call('area/covers', [ $parent], 
+                type => (join ',', @types)
+            );
+        }
+        elsif (@types) {
+
+            if ($types[0] =~/UTA/) {
+                $c->stash->{report_filter} = 'city';
+            }
+            elsif ($types[0] =~ /LBO/) {
+                $c->stash->{report_filter} = 'london';
+            }
+            elsif ($types[0] =~ /CTY/) {
+                $c->stash->{report_filter} = 'dc';
+            }
+
+            mySociety::MaPit::call('areas', [ map uc, @types]);
+        }
+    };
+    if ($areas) {
+        my @ids = keys %$areas;
+        $rs = $rs->search({ id => \@ids });
```

Comment:
This code only works if body ID == area ID which isn't always the case. This either needs documenting as with other places this occurs, changing so that it uses the `body_area`s table, or putting in the UK cobrand where it currently is the case.